orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/model/Stage.java

```diff
@@ -384,81 +369,76 @@ public Task taskById(String taskId) {
    */
   public List<Stage> directAncestors() {
     if (execution != null) {
-      Set<String> visited = Sets.newHashSetWithExpectedSize(execution.getStages().size());
-      return ImmutableList.<Stage>builder()
-          .add(this)
-          .addAll(getAncestorsImpl(visited, true))
-          .build();
-    } else {
-      return emptyList();
+      return getAncestorsImpl(true);
     }
+
+    return emptyList();
   }
 
   /**
    * Worker method to get the list of all ancestors (parents and, optionally, prerequisite stages)
    * of the current stage.
    *
-   * @param visited list of visited nodes
    * @param directParentOnly true to only include direct parents of the stage, false to also include
    *     stages this stage depends on (via requisiteRefIds)
    * @return list of ancestor stages
    */
-  private List<Stage> getAncestorsImpl(Set<String> visited, boolean directParentOnly) {
-    visited.add(this.refId);
+  public ImmutableList<Stage> getAncestorsImpl(boolean directParentOnly) {
```

Comment:
I believe this can be `private`

Reply:
ah, yes, changed it for testing and forgot to undo, thank you!